From c612eb9b7600b26e1bb5b26bb486b3d4225e6275 Mon Sep 17 00:00:00 2001
From: Archana-Shinde1 <archana.shinde1@ibm.com>
Date: Tue, 2 Sep 2025 12:29:41 +0000
Subject: [PATCH] BUILD changes for external sqlite package

---
 third_party/systemlibs/sqlite.BUILD | 26 +++++++++++++++++++++++++-
 1 file changed, 25 insertions(+), 1 deletion(-)

diff --git a/third_party/systemlibs/sqlite.BUILD b/third_party/systemlibs/sqlite.BUILD
index 88a84a96137..973d2ff2a07 100644
--- a/third_party/systemlibs/sqlite.BUILD
+++ b/third_party/systemlibs/sqlite.BUILD
@@ -1,12 +1,36 @@
 licenses(["unencumbered"])  # Public Domain
 
+HEADERS = [
+   "sqlite3.h",
+   "sqlite3ext.h",
+]
+
+LIBS = [
+   "libsqlite3.so",
+   "libsqlite3.so.0",
+]
+
+
 # Production build of SQLite library that's baked into TensorFlow.
 cc_library(
     name = "org_sqlite",
-    linkopts = ["-lsqlite3"],
+    hdrs = HEADERS,
+    srcs = LIBS,
+    includes = ["."],
     visibility = ["//visibility:public"],
 )
 
+genrule(
+    name = "sqlite-files",
+    outs = HEADERS + LIBS,
+    cmd = """
+      cp -fL "$(INCLUDEDIR)/sqlite3.h" "$(@D)" &&
+      cp -fL "$(INCLUDEDIR)/sqlite3ext.h" "$(@D)" &&
+      cp -fL "$(LIBDIR)/libsqlite3.so.0" "$(@D)" &&
+      ln -sf "libsqlite3.so.0" "$(@D)/libsqlite3.so"
+    """,
+)
+
 # This is a Copybara sync helper for Google.
 py_library(
     name = "python",
-- 
2.40.1

